tests/tests_unit/test_credential_providers.py

```diff
@@ -67,3 +67,24 @@ def test_access_token_expired(self, mock_oauth_session, mock_backend_client):
         creds = OAuthClientCredentials(**self.DEFAULT_PROVIDER_ARGS)
         assert "Authorization", "Bearer azure_token_expired" == creds.authorization_header()
         assert "Authorization", "Bearer azure_token_refreshed" == creds.authorization_header()
+
+
+class TestOAuthClientCertificate:
+    DEFAULT_PROVIDER_ARGS = {
+        "authority_url": "https://login.microsoftonline.com/xyz",
+        "client_id": "azure-client-id",
+        "cert_thumbprint": "XYZ123",
+        "certificate": "certificatecontents123",
+        "scopes": ["https://greenfield.cognitedata.com/.default"],
+    }
+
+    @patch("cognite.client.credentials.ConfidentialClientApplication")
+    def test_access_token_generated(self, mock_msal_app):
+        mock_msal_app().return_value = Mock()
+        mock_msal_app().acquire_token_for_client.return_value = {
+            "access_token": "azure_token",
+            "expires_in": 1000,
+        }
+        creds = OAuthClientCertificate(**self.DEFAULT_PROVIDER_ARGS)
+        creds._refresh_access_token()
+        assert "Authorization", "Bearer azure_token" == creds.authorization_header()
```

Comment:
This is also redundant since `authorization_header()` invokes a refresh.
```suggestion
        creds = OAuthClientCertificate(**self.DEFAULT_PROVIDER_ARGS)
        assert "Authorization", "Bearer azure_token" == creds.authorization_header()
```

Reply:
Removed redundant line.